pkg/planner/core/logical_plan_builder.go

```diff
@@ -940,19 +940,30 @@ func (b *PlanBuilder) buildSelection(ctx context.Context, p base.LogicalPlan, wh
 	for _, expr := range expressions {
 		cnfItems := expression.SplitCNFItems(expr)
 		for _, item := range cnfItems {
```

Comment:
Could you please add a todo comment in here ? We prepare to move those kind of TableDual rewrite rule into our logical optimizer not here.
Or if it is ok to you to create a new logical rule and move those logic into new rule ?